grub-core/loader/i386/linux.c

```diff
@@ -543,6 +694,9 @@ grub_linux_boot (void)
 		(unsigned) ctx.real_size,
 		(unsigned) efi_mmap_size);
 
+  if (grub_slaunch_platform_type () == SLP_INTEL_TXT)
+    ap_wake_block_size = GRUB_MLE_AP_WAKE_BLOCK_SIZE;
```

Comment:
Few lines lower we need to allocate more memory for the AP wake block:

```
    if (grub_add (ctx.real_size, efi_mmap_size, &sz))
      return GRUB_ERR_OUT_OF_RANGE;
```

change to:

```
    if (grub_add (ctx.real_size, efi_mmap_size + ap_wake_block_size, &sz))
      return GRUB_ERR_OUT_OF_RANGE;
```

Reply:
@miczyg1 changed https://github.com/TrenchBoot/grub/pull/8/commits/f228aee64e622d063b08b31941c50b38b97672ee#diff-8c69f5f8f99bf5f4195720ea867e6e8fe273f9261ce151f58222e0d73b34c208R707